src/pages/RFOX/components/Stake/hooks/useRfoxStake.tsx

```diff
@@ -129,13 +130,13 @@ export const useRfoxStake = ({
     return encodeFunctionData({
       abi: erc20Abi,
       functionName: 'approve',
-      args: [RFOX_PROXY_CONTRACT, BigInt(amountCryptoBaseUnit)],
+      args: [getStakingContract(stakingAssetId), BigInt(amountCryptoBaseUnit)],
```

Comment:
nitpick: `stakingContractByAssetId[stakingAssetId]` should be enough without the need to resort to this method, since trying to build/broadcast with `undefined` first arg will revert and throw

Reply:
ditto for `spendeer` and `to` below

Reply:
why not throw early on data we know is incorrect than possible invalid tx built? I feel like this makes more sense imo and we can handle errors more explicitely if needed as well or have better error logging to know what data is being used causing the throw

Reply:
Happy with both solutions!